controllers/scheduler.go

```diff
@@ -38,8 +38,11 @@ type Scheduler struct {
 	Log    logr.Logger
 }
 
-// Only reconciles Works that are from a Promise Dependency
-func (s *Scheduler) ReconcileDestination() error {
+// Reconciles all Works by scheduling each Work's WorkloadGroups to appropriate
+// Destinations.
+// Only reconciles Works that are from a Promise Dependency.
+// This is called by the Destination controller to schedule Workloads to Destinations.
```

Comment:
>// This is called by the Destination controller to schedule Workloads to Destinations.

is this comment (calling out who calls this function) necessary? it feels like one of those comments that can quickly get outdated and it's simple enough to find callers using any IDE.

Reply:
Yep, you're right - will delete! (I initially put these comments in to help me understand the flow better)